### DISCUSSION TITLE
Model Outline

### DISCUSSION THREAD

**Discussion by pusia1** [pelicanmapping/osgearth#2327] [Support Q&A]
I want show osgEarth::ModelNode Outline.

I tried like this. 

but did not apply.

`
		m_rpModelNode = new osgEarth::ModelNode(pMapNode, style);
		const char* VS =
			"#version 330 compatibility \n"
			"uniform float outlineWidth; \n"
			"void main()"
			"{ \n"
			"    vec4 clip = gl_ModelViewProjectionMatrix * gl_Vertex; \n"
			"    vec3 normal = normalize(gl_NormalMatrix * gl_Normal); \n"
			"    normal.x *= gl_ProjectionMatrix[0][0]; \n"
			"    normal.y *= gl_ProjectionMatrix[1][1]; \n"
			"    clip.xy += normal.xy * clip.w * outlineWidth;\n"
			"    gl_Position = clip; \n"
			"} \n";

		const char* FS =
			"#version 330\n"
			"uniform vec4 outlineColor; \n"
			"void main() \n"
			"{ \n"
			"    gl_FragColor = outlineColor; \n"
			"} \n";
		osg::Program* program = new osg::Program();
		program->addShader(new osg::Shader(osg::Shader::VERTEX, VS));
		program->addShader(new osg::Shader(osg::Shader::FRAGMENT, FS));

		osg::Group* outline = new osg::Group();
		osg::StateSet* outlineSS = outline->getOrCreateStateSet();
		outlineSS->setRenderBinDetails(1, "RenderBin");
		outlineSS->setAttributeAndModes(new osg::Depth(osg::Depth::LESS, 0.0, 1.0, false));
		outlineSS->setMode(GL_CULL_FACE, 0);
		outlineSS->setAttribute(program);
		outlineSS->addUniform(new osg::Uniform("outlineWidth", 0.002f));
		outlineSS->addUniform(new osg::Uniform("outlineColor", osg::Vec4f(0.0f, 0.7f, 0.0f, 1.0f)));
		outline->addChild(rpModel);

		osg::ref_ptr<osg::Group> rpRenderModel = new osg::Group();
		rpRenderModel->getOrCreateStateSet()->setRenderBinDetails(2, "RenderBin");
		rpRenderModel->addChild(m_rpModelNode);
		

		rpRoot->addChild(rpRenderModel);
		rpRoot->addChild(outline);
`

How can I Show Model Outline?

osg version 3.6.0
osgEarth version 3.4

**Comment by gwaldron** [comment]
Does your code work with a normal OSG model? (outside of osgEarth?)

**Reply by pusia1** [reply]
you're right.
When using only osg without using osgEarth, the line is visible normally.
